src/validators/OidcRecoveryValidator.sol

```diff
@@ -17,48 +17,127 @@ import { Groth16Verifier } from "../autogenerated/JwtTxValidationVerifier.sol";
+  /// @param iss The OIDC issuer.
+  /// @param isNew True if the OIDC key is new, false if it is an update.
   event OidcKeyUpdated(address indexed account, bytes32 oidcDigest, string iss, bool isNew);
+
+  /// @notice Emitted when an OIDC account is deleted.
+  /// @param account The address of the account that deleted the OIDC account.
+  /// @param oidcDigest The PoseidonHash(sub || aud || iss || salt) of the OIDC key.
   event OidcAccountDeleted(address indexed account, bytes32 oidcDigest);
 
+  /// @notice Thrown when calling `validateSignature` since it is not implemented.
+  error ValidateSignatureNotImplemented();
+
+  /// @notice Thrown when an address is not found for a given OIDC digest.
+  /// @param digest The OIDC digest.
+  error AddressNotFoundForDigest(bytes32 digest);
+
+  /// @notice Thrown when trying to add an OIDC account with an OIDC digest that is already registered in another account.
+  /// @param digest The OIDC digest.
+  error OidcDigestAlreadyRegisteredInAnotherAccount(bytes32 digest);
+
+  /// @notice Thrown when there is no OIDC data for a given address.
+  /// @param account The address.
+  error NoOidcDataForGivenAddress(address account);
+
+  /// @notice Thrown when the zk proof verification fails.
+  error ZkProofVerificationFailed();
+
+  /// @notice Thrown when the web authentication validator address is invalid.
+  /// @param expectedValidator The expected address of the web authentication validator.
+  /// @param validator The address of the web authentication validator.
+  error InvalidWebAuthnValidatorAddress(address expectedValidator, address validator);
+
+  /// @notice Thrown when the transaction to address overflows.
+  error TransactionToOverflow();
+
+  /// @notice Thrown when trying to validate a non-function call transaction.
+  error OnlyFunctionCallsSupported();
+
+  /// @notice Thrown when trying to validate a transaction that does not call `addValidationKey` on the `WebAuthValidator` contract.
+  /// @param expectedSelector The expected selector of the `addValidationKey` function.
+  /// @param actualSelector The actual selector of the function being called.
+  error UnauthorizedFunctionCall(bytes4 expectedSelector, bytes4 actualSelector);
+
+  /// @notice Thrown when the passkey hash is invalid.
+  /// @param expectedHash The expected passkey hash.
+  /// @param actualHash The actual passkey hash.
+  error InvalidPasskeyHash(bytes32 expectedHash, bytes32 actualHash);
+
+  /// @notice Thrown when the account is not ready to recover.
+  error NotReadyToRecover();
+
+  /// @notice The data for an OIDC account.
+  /// @param oidcDigest The PoseidonHash(sub || aud || iss || salt) of the OIDC key.
+  /// @param iss The OIDC issuer.
+  /// @param readyToRecover True if the account is ready to recover, false otherwise.
+  /// @param pendingPasskeyHash The hash of the pending passkey.
+  /// @param recoverNonce The nonce for the recovery process to prevent replay attacks.
```

Comment:
Same here. A little bit more of explanation.

`This prevents the same zk proof to be used more than once. The value is used to build the jwt nonce, and gets incremented each time a zk proof is successfully verified.`